File: sso/src/main/java/com/a2m/sso/model/DataResponse.java
package com.a2m.sso.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * Author tiennd
 * Created date 2023-07-08
 */
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@JsonInclude(value = JsonInclude.Include.NON_NULL)
public class DataResponse<T> {
    private String status;
    private String message;
    private T responseData;
}

File: a2m-front/src/app/_model/resp/data-response.ts
export class DataResponse {
    status: string;
    message: string;
    listResponseData: Object[];
    responseData: Object;
    constructor(status: string, message: string, responseData: Object, listResponseData: Object[]) {
        this.status = status;
        this.message = message;
        this.responseData = responseData;
        this.listResponseData = listResponseData;
    }
}
